### PR TITLE
[CARBONDATA-2313] Fixed SDK writer issues and added comment for SDK writer API methods

### PR DESCRIPTION
**Pull request by ajantha-bhat**
*Blockletsize and Blocksize issue fix in sdk writer
*Decimal dataype issue fix in sdk writer
*Drop table issue in cluster
*Added comment for SDK writer API methods
*Query two writer's output at same path issue fix
*Block alter table rename

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NO
 
 - [ ] Any backward compatibility impacted? NO
 
 - [ ] Document update required? NO

 - [ ] Testing done. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3629/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4320/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4851/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java [hunk 1]
```diff
@@ -58,44 +58,85 @@
   private boolean isUnManagedTable;
   private long UUID;
 
+  /**
+   * prepares the builder with the schema provided
+   * @param schema
```
Comment (jackylk) [inline store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java]: provide description for @param and @return

Comment (ajantha-bhat) [reply store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java]: done. Added.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4868/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3646/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4337/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3654/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4877/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4344/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4345/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3667/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4890/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3704/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4925/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4388/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3710/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4929/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4952/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4409/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by ajantha-bhat** [comment]
@jackylk & @gvramana : please review this PR.

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4435/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4990/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3771/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3778/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4995/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4439/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5004/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3788/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3818/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5034/

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java [hunk 2]
```diff
@@ -337,11 +351,12 @@ public static Object getMeasureDefaultValueByType(ColumnSchema columnSchema,
    * @param blockExecutionInfo
    * @param queryMeasures        measures present in query
    * @param currentBlockMeasures current block measures
+   * @param queryModel carbonQueryModel
    * @return measures present in the block
    */
   public static List<ProjectionMeasure> createMeasureInfoAndGetCurrentBlockQueryMeasures(
       BlockExecutionInfo blockExecutionInfo, List<ProjectionMeasure> queryMeasures,
-      List<CarbonMeasure> currentBlockMeasures) {
+      List<CarbonMeasure> currentBlockMeasures, QueryModel queryModel) {
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: Pass flag, dont pass all the querymodel, Restructuring doesnot have any relation with Querymodel

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java [hunk 3]
```diff
@@ -354,7 +369,13 @@ public static Object getMeasureDefaultValueByType(ColumnSchema columnSchema,
       // then setting measure exists is true
       // otherwise adding a default value of a measure
       for (CarbonMeasure carbonMeasure : currentBlockMeasures) {
-        if (carbonMeasure.getColumnId().equals(queryMeasure.getMeasure().getColumnId())) {
+        // If it is unmanaged table just check the column names, no need to validate column id as
+        // multiple sdk's output placed in a single folder doesn't have same column ID but can
+        // have same column name
+        if (carbonMeasure.getColumnId().equals(queryMeasure.getMeasure().getColumnId()) ||
+            ((queryModel != null) && (queryModel.getTable().getTableInfo().isUnManagedTable()) &&
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: remove null check and add method isColumnMatches

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestUnmanagedCarbonTable.scala [hunk 4]
```diff
@@ -266,16 +270,23 @@ class TestUnmanagedCarbonTable extends QueryTest with BeforeAndAfterAll {
...
+    //13. Alter table rename command
+    exception = intercept[MalformedCarbonCommandException] {
+      sql("ALTER TABLE sdkOutputTable RENAME to newTable")
+    }
+    assert(exception.getMessage()
+      .contains("Unsupported operation on unmanaged table"))
 
     sql("DROP TABLE sdkOutputTable")
-    // drop table should not delete the files
+     //drop table should not delete the files
```
Comment (gvramana) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestUnmanagedCarbonTable.scala]: correct indent

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala [hunk 5]
```diff
@@ -73,22 +73,27 @@ private[sql] case class CarbonAlterTableRenameCommand(
                    s"Table $oldDatabaseName.$oldTableName does not exist")
       throwMetadataException(oldDatabaseName, oldTableName, "Table does not exist")
     }
+
+    var carbonTable: CarbonTable = null
+    carbonTable = metastore.lookupRelation(Some(oldDatabaseName), oldTableName)(sparkSession)
```
Comment (gvramana) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala]: This cannot be moved outside lock

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala [hunk 6]
```diff
@@ -565,7 +563,9 @@ class CarbonFileMetastore extends CarbonMetaStore {
             tableModifiedTimeStore.get(CarbonCommonConstants.DATABASE_DEFAULT_NAME))) {
         metadata.carbonTables = metadata.carbonTables.filterNot(
           table => table.getTableName.equalsIgnoreCase(tableIdentifier.table) &&
-        table.getDatabaseName.equalsIgnoreCase(tableIdentifier.database.getOrElse("default")))
+                   table.getDatabaseName.equalsIgnoreCase(
```
Comment (gvramana) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala]: Why it coming here for unmanged table

Comment (ajantha-bhat) [reply integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala]: Ran all the test case. It is not coming now. Removed this changes

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala (hunk 5 above)
Comment (ajantha-bhat) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala]: OK. modified

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestUnmanagedCarbonTable.scala (hunk 4 above)
Comment (ajantha-bhat) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestUnmanagedCarbonTable.scala]: done

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 2 above)
Comment (ajantha-bhat) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: OK

File: core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java (hunk 3 above)
Comment (ajantha-bhat) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: ok

Comment (ajantha-bhat) [reply core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java]: removed null check. cannot call from a method as one is for measure type and other is a dimension type.

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5049/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5051/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5052/

**Comment by ajantha-bhat** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5053/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3835/

**Comment by ajantha-bhat** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3839/

**Comment by ajantha-bhat** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5061/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3844/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5066/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3856/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5077/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5080/

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**